scripts/build-zephyr.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+if [ "`whoami`" = "root" ]
+then
+	echo "Running the script as root is not permitted"
+	exit 1
+fi
+
+CALLED=$_
+[[ "${BASH_SOURCE[0]}" != "${0}" ]] && SOURCED=1 || SOURCED=0
+
+SCRIPT_SRC=$(realpath ${BASH_SOURCE[0]})
+SCRIPT_DIR=$(dirname $SCRIPT_SRC)
+TOP_DIR=$(realpath $SCRIPT_DIR/..)
+
+if [ $SOURCED = 1 ]; then
+	echo "You must run this script, rather then try to source it."
+	echo "$SCRIPT_SRC"
+	exit 1
+fi
+
+if [ -z "$HDMI2USB_ENV" ]; then
+	echo "You appear to not be inside the HDMI2USB environment."
+	echo "Please enter environment with:"
+	echo "  source scripts/enter-env.sh"
+	exit 1
+fi
+
+if [ -z "$ZEPHYR_SDK_INSTALL_DIR" ]; then
+	echo "Zephyr SDK not found"
```

Comment:
Can you make this script download Zephyr SDK?
Also would it be better if this checked out a git repository like the Linux script does?
IE I think people running this script will probably want to hack on zephyr itself?

Reply:
litex-buildenv should make it a little easier for the end user, and maybe install the ZEPHYR_SDK.

```
$ wget https://github.com/zephyrproject-rtos/meta-zephyr-sdk/releases/download/0.10.0/zephyr-sdk-0.10.0-setup.run
$ chmod +x zephyr-sdk-0.10.0-setup.run
$ ./zephyr-sdk-0.10.0-setup.run
```

https://docs.zephyrproject.org/latest/getting_started/installation_linux.html
https://github.com/zephyrproject-rtos/meta-zephyr-sdk/

Not sure that grabbing the source & building the SDK is the best option.

It's built in a docker, using poky (openembedded)...  You can build without the SDK, but I haven't tried it.

Reply:
I think downloading and unpacking the SDK in `download-env.sh` if no SDK is detected in the system is the simplest option.

Using pre-built SDK lets you to do hacking in Zephyr anyway.

Reply:
Can you download the SDK into the `./build/` directory?

Reply:
The script downloads SDK to `build/zephyr_sdk`.

Reply:
Great!